utils/instructionTools.ts

```diff
@@ -582,29 +584,55 @@ export async function getCreateTokenMetadataInstruction({
   schema,
   form,
   programId,
+  wallet,
   governedMintInfoAccount,
   setFormErrors,
   mintAuthority,
   payerSolTreasury,
+  shouldMakeSolTreasury,
 }: {
   schema: any
   form: any
   programId: PublicKey | undefined
+  wallet: WalletAdapter | undefined
   governedMintInfoAccount: AssetAccount | undefined
   setFormErrors: any
   mintAuthority: PublicKey | null | undefined
   payerSolTreasury: PublicKey | null | undefined
+  shouldMakeSolTreasury: boolean
 }): Promise<UiInstruction> {
   const isValid = await validateInstruction({ schema, form, setFormErrors })
   let serializedInstruction = ''
   const prerequisiteInstructions: TransactionInstruction[] = []
+
+  let payer = payerSolTreasury
+
+  if (!payer && shouldMakeSolTreasury && governedMintInfoAccount) {
+    payer = await getNativeTreasuryAddress(
+      governedMintInfoAccount.governance.owner,
+      governedMintInfoAccount.governance.pubkey
+    )
+  }
+
   if (
     isValid &&
     programId &&
     form.mintAccount?.pubkey &&
     mintAuthority &&
-    payerSolTreasury
+    payer &&
+    wallet
   ) {
+    if (shouldMakeSolTreasury && !payerSolTreasury) {
+      const mintAmount = parseMintNaturalAmountFromDecimal(0.006, 9)
```

Comment:
This should be more specific and done only if needed 
1) call `connection.getMinimumBalanceForRentExemption(0)` to get the exact amount needed for the SOL Treasury account itself (it has 0 size) 
2) call `connection.getMinimumBalanceForRentExemption(sizeof(Metadata))` to get the exact amount for the metadata account
3) `transfer amount `=  `1) + 2) - current balance (SOL treasury)`.  transfer only if positive (not sufficient balance)

Reply:
The metadataSize is hardcoded at this moment but should be caliculated in the future.
(On 8.July.2022, Metadata.getMinimumBalanceForRentExemption is returning wrong price.)

Now I'm asking to the Metaplex team about the functions.